File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/domain/interactor/lock/getter/GetLockConnectionStatus.java
package com.lattis.ellipse.domain.interactor.lock.getter;

import com.lattis.ellipse.domain.executor.PostExecutionThread;
import com.lattis.ellipse.domain.executor.ThreadExecutor;
import com.lattis.ellipse.domain.interactor.lock.base.BaseLockUseCase;
import com.lattis.ellipse.domain.model.Lock;
import com.lattis.ellipse.domain.repository.BluetoothRepository;
import com.lattis.ellipse.domain.repository.LockRepository;

import javax.inject.Inject;

import io.reactivex.Observable;

public class GetLockConnectionStatus extends BaseLockUseCase<Boolean> {

    private Lock lock;

    @Inject
    GetLockConnectionStatus(ThreadExecutor threadExecutor,
                           PostExecutionThread postExecutionThread,
                           BluetoothRepository bluetoothRepository, LockRepository lockRepository) {
        super(threadExecutor, postExecutionThread,bluetoothRepository,lockRepository);
    }


    public GetLockConnectionStatus forLock(Lock lock) {
        this.lock = lock;
        return this;
    }

    @Override
    protected Observable<Boolean> buildUseCaseObservable() {
        return isLockConnected(lock);
    }
}

File: velolabs_repos/lattis_android_v2/domain/src/main/java/com/lattis/domain/usecase/lock/getter/GetLockConnectionStatus.kt
package com.lattis.domain.usecase.lock.getter

import com.lattis.domain.executor.PostExecutionThread
import com.lattis.domain.executor.ThreadExecutor
import com.lattis.domain.repository.BluetoothRepository
import com.lattis.domain.repository.LockRepository
import com.lattis.domain.usecase.lock.base.BaseLockUseCase
import com.lattis.domain.models.Lock
import com.lattis.domain.repository.SaSOrPSLockRepository
import io.reactivex.rxjava3.core.Observable
import javax.inject.Inject

class GetLockConnectionStatus @Inject internal constructor(
    threadExecutor: ThreadExecutor,
    postExecutionThread: PostExecutionThread,
    bluetoothRepository: BluetoothRepository, lockRepository: LockRepository,
    saSOrPSLockRepository: SaSOrPSLockRepository
) : BaseLockUseCase<Boolean>(
    threadExecutor,
    postExecutionThread,
    bluetoothRepository,
    lockRepository,
    saSOrPSLockRepository
) {
    lateinit var lock: Lock
    fun forLock(lock: Lock): GetLockConnectionStatus {
        this.lock = lock
        return this
    }

    override fun buildUseCaseObservable(): Observable<Boolean> {
        return isLockConnected(lock)
    }
}